### PR TITLE
Jenkinsfile: provision VMs in separate shell

### PR DESCRIPTION
**Pull request by ianvernon**
This decouples testing from provisioning.

Signed-off by: Ian Vernon <ian@cilium.io>

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/6130)

### CODE REVIEW

**Comment by ianvernon** [comment]
test-me-please

**Comment by ianvernon** [comment]
test-me-please

**Comment by ianvernon** [comment]
Monitor tests failed: 
https://jenkins.cilium.io/job/Cilium-PR-Ginkgo-Tests-Validated/7773/

**Comment by ianvernon** [comment]
test-me-please

**Comment by eloycoto** [comment]
test-me-please

**Comment by ianvernon** [comment]
Monitor tests keep failing. There must be some behavior that is skipped when `--cilium.provision=false` is set in the CI. I'll work on fixing this up locally.

**Comment by ianvernon** [comment]
test-me-please

**Comment by ianvernon** [comment]
test-me-please

**Comment by ianvernon** [comment]
test-me-please

**Review by aanm — APPROVED** [review]

File: test/test_suite_test.go [hunk 1]
```diff
@@ -152,10 +152,12 @@ func reportCreateVMFailure(vm string, err error) {
 var _ = BeforeAll(func() {
 	var err error
 
-	if !config.CiliumTestConfig.Reprovision {
-		// The developer has explicitly told us that they don't care
-		// about updating Cilium inside the guest, so skip setup below.
-		return
+	logger := log.WithFields(logrus.Fields{"testName": "BeforeSuite"})
```
Comment (eloycoto) [inline test/test_suite_test.go]: Update the testName to BeforeAll

File: test/test_suite_test.go [hunk 2]
```diff
@@ -152,10 +152,12 @@ func reportCreateVMFailure(vm string, err error) {
...
+	scope, err := helpers.GetScope()
+	if err != nil {
```
Comment (eloycoto) [inline test/test_suite_test.go:157]: Quick one, the lines from 172 to 176 should be removed no?
https://github.com/cilium/cilium/blob/205b1ca5f44efae964c4fca561f615c3b739d5ae/test/test_suite_test.go#L155-L176

**Review by eloycoto — CHANGES_REQUESTED** [review]
Minor thing, It'll be great if we can validate that 
Nightly test are working with these changes too ;-)

**Comment by ianvernon** [comment]
test-me-please

**Comment by ianvernon** [comment]
test-nightly

**Review by eloycoto — APPROVED** [review]

**Comment by eloycoto** [comment]
Nightly is working, happy to merge without finish ;-)

**Pull request merged**